firebase/dbOperations.ts

```diff
@@ -151,11 +151,41 @@ export const deleteSpecialization = async (specializationId: string) =>
 
 /* --- REVIEWS (keyed by courseId-year-semesterId) --- */
 const baseCollectionReviewsData = 'reviewsData'
+const LEN_SIMPLE_COURSE_NUMBER = 5
+const LEN_COMPOUND_COURSE_NUMBER = 6
+
+const parseReviewId = (reviewId: string) => {
+	let courseId = ''
+	let departmentId = ''
+	let courseNumberA = ''
+	let courseNumberB = ''
+	let year = ''
+	let semesterTerm = ''
+
+	const parsedValues = reviewId.split('-')
+
+	if (parsedValues.length === LEN_SIMPLE_COURSE_NUMBER) {
+		;[departmentId, courseNumberA, year, semesterTerm] = parsedValues
+		courseId = `${departmentId}-${courseNumberA}`
+	}
+
+	if (parsedValues.length === LEN_COMPOUND_COURSE_NUMBER) {
+		;[departmentId, courseNumberA, courseNumberB, year, semesterTerm] =
+			parsedValues
+		courseId = `${departmentId}-${courseNumberA}-${courseNumberB}`
+	}
+
+	return {
+		courseId,
+		year,
+		semesterTerm,
+	}
+}
```

Comment:
FYI refactored to extract `courseId`, `year`, and `semester` from the `reviewId` to make the corresponding CRUD function calls less tedious with additional arguments previously required for these fields